PracticeProject/API/UserAPI.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  UserAPI.swift
+//  PracticeProject
+//
+//  Created by Gnanavel on 31/03/22.
+//
+
+import Foundation
+
+class UserAPI : API {
+    
+    private let apiBaseURL = "https://jsonplaceholder.typicode.com"
+    
+    func fetchUsers(excludingUserWithID: String?, success: @escaping (UsersList?) -> Void, failure: @escaping (FetchError?) -> Void) {
```

Comment:
Make sure this `API` implementation always calls back with either `success` or `failure`. 
There shouldn't be a case where both or neither callbacks are invoked.

Reply:
@nets-dev  Yes, It should be invoked as either `success` or `failure`.

Reply:
The code still does not guarantee that either `success` or `failure` is returned in *ALL* scenarios. Inspect your code carefully and find the problem. Once you do, think of ways on how you can guarantee such mistake doesn't happen (by you or other maintainers of the codebase) in the future.